packages/devtools_app/lib/src/profiler/cpu_profile_model.dart

```diff
@@ -407,3 +423,47 @@ int stackFrameIdCompare(String a, String b) {
     throw error;
   }
 }
+
+class CpuProfileStore {
+  final _store = <TimeRange, CpuProfileData>{};
+
+  CpuProfileData storedProfile(TimeRange time) {
+    // If we have a profile for a time range encompassing [time], then we can
+    // generate and cache the profile for [time] without needing to pull data
+    // from the vm service.
+    _maybeGenerateSubProfile(time);
+    return _store[time];
+  }
+
+  void storeProfile(TimeRange time, CpuProfileData profile) {
```

Comment:
nit: maybe call this method just `add` for consistency with the Dart collection types. `addProfile` or `saveProfile` would also be fine.

Reply:
done